go/internal/cryptoutil/cryptoutil_test.go

```diff
@@ -0,0 +1,48 @@
+package cryptoutil
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+)
+
+func TestSeedFromEd25519PrivateKey(t *testing.T) {
+	priv, _, _ := crypto.GenerateECDSAKeyPair(rand.Reader)
+	_, err := SeedFromEd25519PrivateKey(priv)
+	if err == nil {
+		t.Error("Shoulf fail with ErrInvalidInput")
```

Comment:
```suggestion
		t.Error("Should fail with ErrInvalidInput")
```

btw, you can use our helpers from https://pkg.go.dev/berty.tech/berty/v2/go/pkg/errcode?tab=doc#ErrCode.Code to compare the code instead of just checking if err is nil

Reply:
Thanks for the review I have updated the PR.